tools/repack-kernel

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
+
+# Author: Maciej Borzecki
+# Modified by Ian Johnson <ian.johnson@canonical.com>
+
+set -e
+
+if [[ -n "$D" ]]; then
+    set -x
+fi
+
+show_help() {
+    echo "usage: repack-kernel <command> <opts>"
+    echo ""
+    echo "handle extraction of the kernel snap to a workspace directory, and later"
+    echo "repacking it back to a snap."
+    echo ""
+    echo "   setup                          - setup system dependencies"
+    echo "   extract <snap-file> <target>   - extract under <target> workspace tree"
+    echo "   prepare <target>               - prepare initramfs & kernel for repacking"
+    echo "   pack <target>                  - pack the kernel"
+    echo "   cull-firmware <target>         - remove unnecessary firmware"
+    echo "   cull-modules <target>          - remove unnecessary modules"
+    echo ""
+}
+
+setup_deps() {
+    if [[ "$UID" != "0" ]]; then
+        echo "run as root (only this command)"
+        exit 1
+    fi
+
+    # carries ubuntu-core-initframfs
+    add-apt-repository ppa:snappy-dev/image -y
+    apt update -y
+    apt install ubuntu-core-initramfs -y
```

Comment:
if this just works in ubuntu then we should exit in case apt is not available and show the corresponding error message.